### PR TITLE
project: Recipe app

### PR DESCRIPTION
**Pull request by taher-software**
In this project , we've collaborated to implement the following features:
- A login page.
- A registration page.
- A food list (with all CRUD implementation, except for 'update').
- A recipes list (with all CRUD implementation, except for 'update').
- A general shopping list view (all missing food for all your recipes and total price).
- A list of all public recipes from other users with their names and total prices.

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+  create_table "foods", force: :cascade do |t|
+    t.string "name"
+    t.string "measurement_unit"
+    t.float "price"
+    t.bigint "user_id", null: false
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+    t.index ["user_id"], name: "index_foods_on_user_id"
+  end
```
Comment (MahmoudBakr23) [inline db/schema.rb:25]: - [ ] Great work so far setting up your database. However, this table here seems to be missing its migration file. Please fix that issue in order for other non-tech reviewers who want to run your app locally according to your step-by-step instructions in the README file. 🙏 😄

File: README.md [hunk 2]
```diff
@@ -1,24 +1,61 @@
...
+
+## Acknowledgments
+
+- Hat tip to anyone whose code was used
+- Inspiration
+- etc
+
+## 📝 License
+
+This project is [MIT](./MIT.md) licensed.
```
Comment (MahmoudBakr23) [inline README.md]: - [ ] Please fix the MIT link for a completely professional repo

File: README.md [hunk 3]
```diff
@@ -1,24 +1,61 @@
-# README
+# Recipe-app
 
-This README would normally document whatever steps are necessary to get the
-application up and running.
+> The Recipe app keeps track of all your recipes, ingredients, and inventory. It will allow you to save ingredients, keep track of what you have, create recipes, and generate a shopping list based on what you have and what you are missing from a recipe. Also, since sharing recipes is an important part of cooking the app allow you to make them public so anyone can access them.
```
Comment (MahmoudBakr23) [inline README.md:4]: - [ ] **OPTIONAL**
It'd be great if you included a couple of images of your app in the README file for a nicer appearance. Also, a live demo link would be even greater 😉

File: test/controllers/public_recipes_controller_test.rb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
+
+class PublicRecipesControllerTest < ActionDispatch::IntegrationTest
+  # test "the truth" do
+  #   assert true
+  # end
```
Comment (MahmoudBakr23) [inline test/controllers/public_recipes_controller_test.rb:6]: - [ ] Please work on testing your controllers accordingly and remove any unnecessary/commented code.

**Review by MahmoudBakr23 — CHANGES_REQUESTED** [review]
Hello team😃, 

Great work and effort you put into this project so far.

## Highlights 🚀 

There are some excellent points that you have made perfectly. Keep up! 💪
- Linters are correct
- Professional PR headline, summary, and commit messages
- Well done setting up the models and controllers
- Nice design following
- Great use of rails methods to manage the queries requested from the database to reduce n+1 problems
- Well done using partials in your views

However, there is always room for improvement.

> Keep in mind that your long hours of work are appreciated. However, the code review process is for improving the code you wrote, and not for criticizing it. 😄

 ## Required Changes ♻️

- [ ] Please check the codebase of your app and gitignore any large/unneeded folders because it's making it harder to review.
- [ ] Kindly improve your tests of this app and add more test cases to your models/integration and unit tests.

_Please check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

_Please see my comments down below!_

Cheers and Happy coding!👏👨‍💻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by t-yanick — CHANGES_REQUESTED** [review]
Hi @Taher-web-dev, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments below and under the review._

### I leave the comments here, it seems there are a lot of unnecessary files in your repository that is causing the browser to freeze and making it very difficult for me to leave inline comments. As requested by the previous code reviewer, please make sure you add unnecessary files to .gitignore. ###

- [ ] As requested by the previous code reviewer, please kindly fix the link to the MIT license, to fix it, create a file in the root of your repository and name it MIT.md, copy the MIT license and paste in to the file, save it, and maintain the same link to the file like you did before the first review. 

- [ ] As requested by the previous code reviewer, please kindly add the migration file for the `foods table`, as this will be of help to non technical users, and some of these are recruiters. 

- [ ] As requested by the previous code reviewer, please more tests to your controllers as this will reduce the probability of your app breaking in production. 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend yo to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by t-yanick — APPROVED** [review]
Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

- [OPTIONAL] As requested by the previous code reviewer, please kindly fix the link to the MIT license, to fix it, create a file in the root of your repository and name it MIT.md, copy the MIT license and paste in to the file, save it, and maintain the same link to the file like you did before the first review.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**